web-registry/src/components/organisms/MediaForm/MediaForm.tsx

```diff
@@ -0,0 +1,149 @@
+
+import type { MediaValuesSimple, MediaErrorsSimple } from './MediaFormSimple';
+import type { MediaErrorsLegacy, MediaValuesLegacy } from './MediaFormLegacy';
+import type { ShaclGraphByUriQuery, Maybe } from '../../../generated/graphql';
+
+export interface MediaBaseValues {
+  'regen:previewPhoto'?: UrlType;
+  'regen:galleryPhotos'?: UrlList;
+  'regen:videoURL'?: UrlType;
+}
+
+type ValueObject = { '@value': string };
+export interface MediaBaseErrors {
+  'regen:previewPhoto'?: ValueObject;
+  'regen:videoURL'?: ValueObject;
+}
+
+export type MediaValues = MediaValuesSimple | MediaValuesLegacy;
+export type MediaErrors = MediaErrorsSimple | MediaErrorsLegacy;
+
+export function isSimpleValues(
+  _values: MediaValues,
+  ccId?: Maybe<string> | string,
+): _values is MediaValuesSimple {
+  return !!ccId;
+}
+
+function isSimpleErrors(
+  _errors: MediaErrors,
+  ccId?: Maybe<string> | string,
+): _errors is MediaErrorsSimple {
+  return !!ccId;
+}
+
+export const cropAspect = { aspect: 322 / 211 }; // px values pulled from mockups (width / height)
+
+interface MediaFormProps {
+  submit: (values: MediaValues) => Promise<void>;
+  onPrev?: () => void;
+  onNext?: () => void;
+  initialValues: MediaValues;
+  creditClassId?: string | null;
+  graphData?: ShaclGraphByUriQuery;
+}
+
+/** Formik Context + handlers for legacy and new media */
+export const MediaForm = ({
+  initialValues,
+  graphData,
+  creditClassId,
+  ...props
+}: MediaFormProps): JSX.Element => {
+  const { confirmSave, isEdit } = useProjectEditContext();
+
+  async function handleSubmit(
+    values: MediaValues,
+    { setSubmitting, setTouched }: FormikHelpers<MediaValues>,
+  ): Promise<void> {
+    setSubmitting(true);
+    try {
+      await props.submit(values);
+      setSubmitting(false);
+      setTouched({}); // reset to untouched
+      if (isEdit && confirmSave) confirmSave();
+    } catch (e) {
+      setSubmitting(false);
+    }
+  }
+
+  const handleValidate = async (values: MediaValues): Promise<MediaErrors> => {
+    const errors: MediaErrors = {};
+    if (graphData?.shaclGraphByUri?.graph) {
+      const projectPageData = {
+        ...getProjectPageBaseData(creditClassId),
+        ...values,
+      };
+      const report = await validate(
+        graphData.shaclGraphByUri.graph,
+        projectPageData,
+        'http://regen.network/ProjectPageMediaGroup',
+      );
+      for (const result of report.results) {
+        const path: string = result.path.value;
+        const compactedPath = getCompactedPath(path) as
+          | keyof MediaValues
+          | undefined;
+        if (compactedPath) {
+          if (
+            [
+              'regen:previewPhoto',
+              'regen:videoURL',
+              'regen:landStewardPhoto',
+            ].includes(compactedPath)
+          ) {
+            errors[compactedPath] = { '@value': requiredMessage };
+          }
+          if (compactedPath === 'regen:galleryPhotos') {
```

Comment:
unfortunately since this field is a @list and the SHACL validation is defined for the elements of this list, there is no way you'll get `compactedPath === 'regen:galleryPhotos'` because there is no constraint per se on this path (but rather on the elements inside)